BUILDING.md

```diff
@@ -531,23 +531,30 @@ make test-only
 
 #### Speeding up frequent rebuilds when developing
 
-If you plan to frequently rebuild Node.js, especially if using several branches,
-installing `ccache` can help to greatly reduce build times. Set up with:
+Tips: The `ccache` utility is widely used and should generally work fine.
+If you encounter any difficulties, consider disabling `mold` as a
+troubleshooting step.
+
+If you plan to frequently rebuild Node.js, especially if using several
+branches, installing `ccache` and `mold` can help to greatly reduce build
+times. Set up with:
 
 On GNU/Linux:
 
 ```bash
-sudo apt install ccache   # for Debian/Ubuntu, included in most Linux distros
-export CC="ccache gcc"    # add to your .profile
-export CXX="ccache g++"   # add to your .profile
+sudo apt install ccache mold   # for Debian/Ubuntu, included in most Linux distros
+export CC="ccache gcc"         # add to your .profile
+export CXX="ccache g++"        # add to your .profile
+export LDFLAGS="-fuse-ld=mold" # add to your .profile
 ```
 
 On macOS:
 
 ```bash
-brew install ccache      # see https://brew.sh
-export CC="ccache cc"    # add to ~/.zshrc or other shell config file
-export CXX="ccache c++"  # add to ~/.zshrc or other shell config file
+brew install ccache mold       # see https://brew.sh
+export CC="ccache cc"          # add to ~/.zshrc or other shell config file
+export CXX="ccache c++"        # add to ~/.zshrc or other shell config file
+export LDFLAGS="-fuse-ld=mold" # add to ~/.zshrc or other shell config file
```

Comment:
> Support for Mach-O targets has been removed.
> See https://github.com/bluewhalesystems/sold for macOS/iOS support.

I think we should use `sold` in macOS

Reply:
@congzhangzh Are you willing to open a PR to replace `mold` to `sold` on mac?

Reply:
Don't forget that sold is not available on homebrew

Reply:
sure, please allow me do more test on mac os 

@rui314 Hi Rui, is that ok we use sold in nodejs,  as this mark on https://bluewhale.systems/ websit
>> **What is the difference between sold and mold?**
>> Other than the branding and their licenses, sold and mold behave the same. Both sold and mold are developed >> by us and our contributors.

btw, and whether mold merge all sold functions?

Reply:
sold is not actively maintained anymore because with Xcode 15, Apple has made their linker much faster than it was. So you want to recommend it instead of sold. IIRC, the Apple's new linker is available as ld-prime after installing Xcode 15.

Reply:
@rui314 tks, can you give more details like export LDFLAGS="-fuse-ld=ld-prime" or do I need full path of ld-prime , as my mac is too old to run xcode now!

Reply:
> sold is not actively maintained anymore because with Xcode 15, Apple has made their linker much faster than it was. So you want to recommend it instead of sold. IIRC, the Apple's new linker is available as ld-prime after installing Xcode 15.

@rui314 @H4ad @anonrig Hi folks, can you help to ensure that the new linker ld-prime works, which point by Rui

Reply:
Since Apple's linker is not my product, and I no longer working on macOS, I don't know whether that LDFLAGS works or not.

Reply:
I don't have macos :/

Also, as mentioned by @anonrig, I think we should remove `mold` from `macos` section and don't use `sold` since is not maintained anymore.

Reply:
ok, get it, I will do a pull request later:)

Reply:
> I don't have macos :/
> 
> Also, as mentioned by @anonrig, I think we should remove `mold` from `macos` section and don't use `sold` since is not maintained anymore.

@H4ad I create a merge request to fix it,  <https://github.com/nodejs/node/pull/52252>